harness/determined/tensorboard/base.py

```diff
@@ -148,7 +148,7 @@ def __init__(
 
         self._work_queue: queue.Queue = queue.Queue(maxsize=work_queue_max_size)
 
-        super().__init__()
+        super().__init__(daemon=True, name="TensorboardUploadThread")
```

Comment:
what is the purpose of adding names to the threads?

Reply:
>Set explicit names for the background threads we spawn to simplify debugging a bit.

whenever you have your program stuck at exit, I like to print the names of the current threads. many other libraries explicitly name their threads so you can see what they do. I think adding a name makes it nicer.